conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -217,6 +217,13 @@ def test_apple_clang_cppstd_flags(self):
         self.assertEqual(_make_cppstd_flag("apple-clang", "14.0", "23"), "-std=c++2b")
         self.assertEqual(_make_cppstd_flag("apple-clang", "14.0", "gnu23"), "-std=gnu++2b")
 
+        self.assertEqual(_make_cppstd_flag("apple-clang", "15.0", "17"), "-std=c++17")
+        self.assertEqual(_make_cppstd_flag("apple-clang", "15.0", "gnu17"), "-std=gnu++17")
+        self.assertEqual(_make_cppstd_flag("apple-clang", "15.0", "20"), "-std=c++20")
+        self.assertEqual(_make_cppstd_flag("apple-clang", "15.0", "gnu20"), "-std=gnu++20")
+        self.assertEqual(_make_cppstd_flag("apple-clang", "15.0", "23"), "-std=c++2b")
+        self.assertEqual(_make_cppstd_flag("apple-clang", "15.0", "gnu23"), "-std=gnu++2b")
```

Comment:
Doesn't AppleClang support `-std=c++23` already?

Reply:
```
bash-3.2$ echo '' | ./clang -x c++ --std=c++23 -E -
error: invalid value 'c++23' in '--std=c++23'
note: use 'c++98' or 'c++03' for 'ISO C++ 1998 with amendments' standard
note: use 'gnu++98' or 'gnu++03' for 'ISO C++ 1998 with amendments and GNU extensions' standard
note: use 'c++11' for 'ISO C++ 2011 with amendments' standard
note: use 'gnu++11' for 'ISO C++ 2011 with amendments and GNU extensions' standard
note: use 'c++14' for 'ISO C++ 2014 with amendments' standard
note: use 'gnu++14' for 'ISO C++ 2014 with amendments and GNU extensions' standard
note: use 'c++17' for 'ISO C++ 2017 with amendments' standard
note: use 'gnu++17' for 'ISO C++ 2017 with amendments and GNU extensions' standard
note: use 'c++20' for 'ISO C++ 2020 DIS' standard
note: use 'gnu++20' for 'ISO C++ 2020 DIS with GNU extensions' standard
note: use 'c++2b' for 'Working draft for ISO C++ 2023 DIS' standard
note: use 'gnu++2b' for 'Working draft for ISO C++ 2023 DIS with GNU extensions' standard
bash-3.2$ ./clang --version
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: x86_64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /Applications/Xcode_15_beta_6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/.
```

See also https://github.com/conan-io/conan/issues/14494#issuecomment-1681625657

Reply:
> ```shell
> ```shell
> $ echo '' | ./clang -x c++ --std=c++23 -E -
> error: invalid value 'c++23' in '--std=c++23'
> note: use 'c++98' or 'c++03' for 'ISO C++ 1998 with amendments' standard
> note: use 'gnu++98' or 'gnu++03' for 'ISO C++ 1998 with amendments and GNU extensions' standard
> note: use 'c++11' for 'ISO C++ 2011 with amendments' standard
> note: use 'gnu++11' for 'ISO C++ 2011 with amendments and GNU extensions' standard
> note: use 'c++14' for 'ISO C++ 2014 with amendments' standard
> note: use 'gnu++14' for 'ISO C++ 2014 with amendments and GNU extensions' standard
> note: use 'c++17' for 'ISO C++ 2017 with amendments' standard
> note: use 'gnu++17' for 'ISO C++ 2017 with amendments and GNU extensions' standard
> note: use 'c++20' for 'ISO C++ 2020 DIS' standard
> note: use 'gnu++20' for 'ISO C++ 2020 DIS with GNU extensions' standard
> note: use 'c++2b' for 'Working draft for ISO C++ 2023 DIS' standard
> note: use 'gnu++2b' for 'Working draft for ISO C++ 2023 DIS with GNU extensions' standard
> $ ./clang --version
> Apple clang version 15.0.0 (clang-1500.0.40.1)
> Target: x86_64-apple-darwin22.6.0
> Thread model: posix
> InstalledDir: /Applications/Xcode_15_beta_6.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/.
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> See also #14494 (comment)
> ```

Thank you. It looks like this was resolved only in Clang 17:

> error: invalid value 'c++27' in '-std=c++27'
note: use 'c++98' or 'c++03' for 'ISO C++ 1998 with amendments' standard
note: use 'gnu++98' or 'gnu++03' for 'ISO C++ 1998 with amendments and GNU extensions' standard
note: use 'c++11' for 'ISO C++ 2011 with amendments' standard
note: use 'gnu++11' for 'ISO C++ 2011 with amendments and GNU extensions' standard
note: use 'c++14' for 'ISO C++ 2014 with amendments' standard
note: use 'gnu++14' for 'ISO C++ 2014 with amendments and GNU extensions' standard
note: use 'c++17' for 'ISO C++ 2017 with amendments' standard
note: use 'gnu++17' for 'ISO C++ 2017 with amendments and GNU extensions' standard
note: use 'c++20' for 'ISO C++ 2020 DIS' standard
note: use 'gnu++20' for 'ISO C++ 2020 DIS with GNU extensions' standard
note: use 'c++23' for 'ISO C++ 2023 DIS' standard
note: use 'gnu++23' for 'ISO C++ 2023 DIS with GNU extensions' standard
note: use 'c++2c' or 'c++26' for 'Working draft for C++2c' standard
note: use 'gnu++2c' or 'gnu++26' for 'Working draft for C++2c with GNU extensions' standard

Reply:
I've checked beta 7 and the result is the same (more or less expected).